fiboa_cli/publish.py

```diff
@@ -0,0 +1,359 @@
+        self.info(f"Validating {parquet_file}")
+        ValidateData().validate(parquet_file, num=-1)
+        self.log("\n  => VALID\n", "success")
+
+        ## Create STAC collection.json
+        self.create_stac_collection(target, file_name, parquet_file, stac_file)
+
+        if generate_meta:
+            self.generate_meta(target, file_name, stac_file)
+
+        pm_file = os.path.join(target, f"{file_name}.pmtiles")
+        if not os.path.exists(pm_file):
+            self.info("Running ogr2ogr | tippecanoe")
+            self.check_command("tippecanoe")
+            self.check_command("ogr2ogr", name="GDAL")
+            self.exc(
+                f"ogr2ogr -t_srs EPSG:4326 -f geojson /vsistdout/ {parquet_file} | tippecanoe -zg --projection=EPSG:4326 -o {pm_file} -l {self.dataset} --drop-densest-as-needed"
+            )
+
+        self.info("Uploading to aws")
+        if not has_write_access:
+            self.info("Get your credentials through the source coop organization.")
+            self.info("Login to AWS Console and generate an access key:")
+            self.info(
+                "  - In AWS console, click on account (right top) press 'Security credentials',"
+            )
+            self.info("  - Go to 'Access keys' and press 'Create access key'")
+            self.info(
+                "  - Run export WS_ACCESS_KEY_ID=<> AWS_SECRET_ACCESS_KEY=<>\n"
+                "    where you copy-past the access key and secret",
+            )
+            self.error("Please set AWS_ env vars for uploading")
+            sys.exit(1)
+
+        self.check_command("aws")
+        self.exc(f"aws s3 sync {target} {self.s3_upload_path}")
+
+    def create_stac_collection(self, target, file_name, parquet_file, stac_file):
+        p_stac = Path(stac_file)
+        if p_stac.exists() and p_stac.stat().st_mtime >= Path(parquet_file).stat().st_mtime:
+            return
+
+        self.success(f"Creating STAC collection.json for {parquet_file}")
+        p_stac.parent.mkdir(exist_ok=True)
+        CreateStacCollection().create_cli(parquet_file, stac_file)
+
+        os.makedirs(os.path.join(target, "stac"), exist_ok=True)
+        data = json.load(open(stac_file, "r"))
+        assert data["id"] == self.dataset, (
+            f"Wrong collection dataset id: {data['id']} != {self.dataset}, for {stac_file}"
+        )
+
+        data["assets"]["data"]["href"] = f"{self.data_url}/{file_name}.parquet"
+
+        if STAC_EXTENSION not in data["stac_extensions"]:
+            data["stac_extensions"].append(STAC_EXTENSION)
+
+        if not any(d.get("rel") == "pmtiles" for d in data["links"]):
+            data["links"].append(
+                {
+                    "href": f"{self.data_url}/{file_name}.pmtiles",
+                    "type": "application/vnd.pmtiles",
+                    "rel": "pmtiles",
+                }
+            )
+
+        with open(stac_file, "w", encoding="utf-8") as f:
+            json.dump(data, f)
```

Comment:
```suggestion
            json.dump(data, f, indent=2)
```

for some nicer JSON output